pandas/core/generic.py

```diff
@@ -2063,6 +2063,7 @@ def to_json(
         index: bool_t = True,
         indent: Optional[int] = None,
         storage_options: StorageOptions = None,
+        mode: str = "w",
```

Comment:
Hmm not sure about `mode` as a keyword - maybe `append_lines` is better? 

I think mode is just confusing since it only works on a very small subset of user cases

Reply:
Sure, I can make this change. My choice of using mode was to keep same naming as previously established (e.g. CSVFormatter(...,mode=mode...) ).

Reply:
mode is appropriate and consistent with other IO methods., but possibly restrict to w|a only, would put before storage_options.

Reply:
Yea I'm just wary of what users expect from this since "mode" is not going to be applicable to the majority of orients. For example, it wouldn't make sense to have `mode="a"` for the table orient because it would produce an invalid document structure that way

In any case can start with this and add tests for the behavior we want to catch

Reply:
Ok, I'll keep mode for now, If in future bigger consensus tell me to change I still can change. From my side, I tend to favor consistency, because of this `mode="a"` sounds better to me.